### PR TITLE
Allow setting namespace

### PR DESCRIPTION
**Pull request by sbquinlan**
## Summary & Motivation

Same as #20691 but properly using default to allow setting the namespace.

## How I Tested These Changes

bk

### CODE REVIEW

**Comment by sbquinlan** [comment]
* **#20764** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20764?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @sbquinlan and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: helm/dagster/charts/dagster-user-deployments/templates/helpers/_helpers.tpl [hunk 1]
```diff
@@ -130,7 +130,7 @@ DAGSTER_K8S_PIPELINE_RUN_ENV_CONFIGMAP: "{{ template "dagster.fullname" . }}-pip
     {{- if .schedulerName }}
     scheduler_name: {{ .schedulerName }}
     {{- end }}
-    namespace: {{ $.Release.Namespace }}
+    namespace: {{ coalesce $.Values.jobNamespace $.Release.Namespace }}
```
Comment (alangenfeld) [inline helm/dagster/charts/dagster-user-deployments/templates/helpers/_helpers.tpl:133]: im not familiar with `coalesce`, you sure this is the right precedence? adding a test would be nice

**Review by alangenfeld — APPROVED** [review]

File: helm/dagster/charts/dagster-user-deployments/templates/helpers/_helpers.tpl (hunk 1 above)
Comment (sbquinlan) [reply helm/dagster/charts/dagster-user-deployments/templates/helpers/_helpers.tpl:133]: Yeah I switched it. Where would tests live for this, sorry I'm rarely in OSS.

Comment (alangenfeld) [reply helm/dagster/charts/dagster-user-deployments/templates/helpers/_helpers.tpl:133]: https://github.com/dagster-io/dagster/tree/master/helm/dagster/schema/schema_tests

the way we have this set-up i think its easy and quick to add

**Pull request merged**